dali/operators/decoder/audio/audio_decoder_op.h

```diff
@@ -54,15 +59,38 @@ class AudioDecoderCpu : public Operator<CPUBackend> {
 
   void RunImpl(workspace_t<Backend> &ws) override;
 
+
   bool CanInferOutputs() const override {
     return true;
   }
 
  private:
-  DALIDataType output_type_;
+  template <typename OutputType>
+  void DecodeSample(
+    const TensorView<StorageCPU, OutputType, 2> &audio,
+    int thread_idx,
+    int sample_idx);
+
+  template <typename OutputType>
+  void DecodeBatch(workspace_t<Backend> &ws);
+
+  int64_t OutputLength(int64_t in_length, double in_rate, int sample_idx) const {
+    if (use_resampling_) {
+      return kernels::signal::resampling::resampled_length(
+          in_length, in_rate, target_sample_rates_[sample_idx]);
+    } else {
+      return in_length;
+    }
+  }
+
+  std::vector<float> target_sample_rates_;
+  kernels::signal::resampling::Resampler resampler_;
```

Comment:
Shouldn't the resampler_ ber per-sample?

Reply:
Not unless per-sample quality is needed - and I have quite strong opinion that it won't and I wouldn't waste memory and cache locality on that assumption unless we see at least potential use case (i.e. if we find it in actual network or at least offline augmentation for that network).